woocommerce/woocommerce-admin
# remove deleted products, coupons from stats when order edited

Fixes #3100

This PR adds code to the coupon & product sync process to remove coupons and/or products from the lookup tables when they have been removed from the orders;

Note: The PR should wait for #2961 to be merged then rebased.

### Detailed test instructions:

- Create an order in the dashboard
- Add 3 different products and a coupon
- Save the order
- Check Tools -> Scheduled Actions to ensure the order is processed
- Edit the order
- Remove the coupon and 2 of the products
- Pay for the order through checkout
- Check Tools -> Scheduled Actions to ensure the order is processed
- Load Analytics -> Orders, Coupons, Products for Today
- All 3 screens should correctly reflect the order items, quantities, and amounts

### Changelog Note:

Fix: Correct product and coupon count on edited orders.
